### DISCUSSION TITLE
Will there be any proper documentation on writing unit tests with TSR?

### DISCUSSION THREAD

**Discussion by abdeliibrahim** [TanStack/router#2742] [Q&A]
Currently there are only a few threads of people working around the lack of testing documentation (particularly with Vitest).

**Comment by schiller-manuel** [comment]
it really depends what you need to test. did you have a look at our internal tests? e.g. 

https://github.com/TanStack/router/blob/main/packages/react-router/tests/useNavigate.test.tsx#L23

is this what you are looking for?
we welcome all contributions so if you want to add something to the documentation, feel free to create a PR!
